src/compiler/scala/tools/nsc/settings/Warnings.scala

```diff
@@ -103,6 +103,7 @@ trait Warnings {
     val StarsAlign             = LintWarning("stars-align",               "Pattern sequence wildcard must align with sequence component.")
     val Constant               = LintWarning("constant",                  "Evaluation of a constant arithmetic expression results in an error.")
     val Unused                 = LintWarning("unused",                    "Enable -Ywarn-unused:imports,privates,locals,implicits.")
+    val NonLocalReturn         = LintWarning("nonlocal-return",           "A return statement used an exception for flow control.")
```

Comment:
The normally-latent grammarian in me notices that, while you're sitting on a camel named "Non Local Return", you're eating a kebab called "nonlocal-return". Not that anyone cares.

Reply:
I checked this afternoon that hyphenated `non-` is British usage; and then it turns on whether it's nonlocal or not a local return, and I went for the former. It's clearly "nonlocal goto"? After deciding on the option name, I was too lazy to tab left for the variable name, but I'll catch it on the next tweak, next week.

Reply:
British people tell me I speak American English, and Americans think I speak British English, so I have no claim to accuracy either way. The clearly optimal choice is backticked command line options.

Reply:
Unfortunately, I did contravene Scala usage in docs & comments. I'm hoping no one notices or cares about the eclectic mix.